### PR TITLE
implements precision recall curves in eval.py

### PR DESCRIPTION
**Pull request by LoganMilandin**
title

### CODE REVIEW

File: eval.py [hunk 1]
```diff
@@ -65,7 +76,7 @@
   transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
 ])
 
-test_labels_csv_path = CSV_BASE_PATH + "CSV NAME HERE"
+test_labels_csv_path = CSV_BASE_PATH + "small.csv"
```
Comment (ThatOneGoat) [inline eval.py:79]: Replace placeholder

File: eval.py [hunk 2]
```diff
@@ -75,8 +86,22 @@
 # =================================================================================================
 # evaluate loaded model on test set
 MISTAKES_SAVE_PATH = "./visualizations/" + SESSION_NAME + "_mistakes.csv"
-test_accuracy, test_loss, cm = evaluate(model, (MODEL_NAME == "inception"), loss_func, test_dataloader, True, MISTAKES_SAVE_PATH,device)
+test_accuracy, test_loss, cm, output_probabilities, corresponding_ground_truths = evaluate(model, (MODEL_NAME == "inception"), loss_func, test_dataloader, True, MISTAKES_SAVE_PATH,device)
```
Comment (ThatOneGoat) [inline eval.py:89]: hmm this function has hella outputs, maybe we should condense them into a tuple or dict or something?

File: utils/training_utils.py [hunk 3]
```diff
@@ -183,7 +183,7 @@ def train(model, num_classes, is_inception, optimizer, scheduler, loss_func, epo
             if is_cyclic_lr:
               scheduler.step()
         
-        correct_preds = torch.where(preds == labels, preds, torch.tensor([-1 for i in preds]).to(device))
+        correct_preds = torch.where(preds == labels, preds, -1)
```
Comment (ThatOneGoat) [inline utils/training_utils.py:186]: No need for the to(device) anymore?

File: utils/training_utils.py [hunk 4]
```diff
@@ -275,7 +280,11 @@ def evaluate(model, is_inception, loss_func, dataset_loader, test, mistakes_save
       # we ignore aux output in test loss calculation
       # since we aren't updating weights
       loss = loss_func(outputs, labels)
-      _, predictions = torch.max(outputs, 1)
+      batch_probabilities = probability_from_outputs(outputs)
```
Comment (ThatOneGoat) [inline utils/training_utils.py:283]: Can you just call the soft max here?

Like nn.Softmax(outputs, dim=1)?

**Review by michaelduan8 — CHANGES_REQUESTED** [review]

File: utils/training_utils.py (hunk 3 above)
Comment (LoganMilandin) [reply utils/training_utils.py:186]: Nope, -1 is a scalar so there's no way to put it on the device

File: utils/training_utils.py (hunk 4 above)
Comment (LoganMilandin) [reply utils/training_utils.py:283]: No, nn.Softmax returns a function

File: utils/training_utils.py (hunk 3 above)
Comment (ThatOneGoat) [reply utils/training_utils.py:186]: whoops, i thought that to(device) was on the outside. looks good to me

**Review by michaelduan8 — APPROVED** [review]

**Comment by jonfroehlich** [comment]
Love this code review team. Rock!
